packages/utils/tests/logger.ts

```diff
@@ -0,0 +1,262 @@
+import { afterEach, describe, expect, jest, test } from '@jest/globals';
+import {
+  createDefaultConsoleLogger,
+  createLoggerWithPrefix,
+  updateLoggerObject,
+} from '../src';
+import * as config from '../src/config';
+
+const info = {
+  message: { message: 'Info Message' },
+  option: { key: 'info' },
+};
+
+const debug = {
+  message: { message: 'Debug Message' },
+  option: { key: 'debug' },
+};
+
+const verbose = {
+  message: { message: 'Verbose Message' },
+  option: { key: 'verbose' },
+};
+
+const warn = {
+  message: { message: 'Warn Message' },
+  option: { key: 'warn' },
+};
+
+const error = {
+  message: { message: 'Error Message' },
+  option: { key: 'error' },
+};
+
+jest.mock('../src/config', () => ({
+  __esModule: true,
+  config: {
+    LOG_LEVEL: 'debug',
+  },
+}));
+
+const mockedConfig = config as jest.Mocked<typeof config>;
+
+describe('Logger', () => {
+  afterEach(() => {
+    jest.clearAllMocks();
+    jest.resetAllMocks();
+  });
+
+  describe('createDefaultConsoleLogger', () => {
+    test('should log messages with meta based on log level priority', () => {
+      const actualConsole = global.console;
+
+      console.info = jest.fn();
+      console.debug = jest.fn();
+      console.warn = jest.fn();
+      console.error = jest.fn();
+
+      const logger = createDefaultConsoleLogger('test');
+
+      logger.info(info.message, info.option);
+      logger.debug(debug.message, debug.option);
+      logger.verbose(verbose.message, verbose.option);
+      logger.warn(warn.message, warn.option);
+      logger.error(error.message, error.option);
+
+      const logDefaultMeta = {
+        service: 'test',
+        timestamp: expect.any(Date),
+      };
+
+      expect(console.info).toHaveBeenCalledWith(info.message, {
+        ...info.option,
+        ...logDefaultMeta,
+      });
+      expect(console.debug).toHaveBeenNthCalledWith(1, debug.message, {
+        ...debug.option,
+        ...logDefaultMeta,
+      });
+      expect(console.debug).toHaveBeenNthCalledWith(2, verbose.message, {
+        ...verbose.option,
+        ...logDefaultMeta,
+      });
+      expect(console.warn).toHaveBeenCalledWith(warn.message, {
+        ...warn.option,
+        ...logDefaultMeta,
+      });
+      expect(console.error).toHaveBeenCalledWith(error.message, {
+        ...error.option,
+        ...logDefaultMeta,
+      });
+
+      console.info = actualConsole.info;
+      console.debug = actualConsole.debug;
+      console.warn = actualConsole.warn;
+      console.error = actualConsole.error;
+    });
+
+    test('should log messages without meta based on log level priority', () => {
+      const infoLogger = jest.spyOn(console, 'info');
```

Comment:
Use `spyOn` or `jest.fn()` to monitor console function calls. Here we are using both methods.

Reply:
using `jest.fn()`